android/src/main/java/com/tailscale/ipn/App.kt

```diff
@@ -265,18 +243,6 @@ class App : Application(), libtailscale.AppContext {
     return null
   }
 
-  // attachPeer adds a Peer fragment for tracking the Activity
-  // lifecycle.
-  fun attachPeer(act: Activity) {
```

Comment:
aha. I see, requestVpnPermission calls VpnService.prepare, so when App.prepareVpn calls VpnService.prepare, this is almost always null (vpn has been prepared), so we never need to call startActivityForResult right now. but if we're planning on moving requestVpnPermission() out of onStart, then we would still need this (and to actually call this function too)

Reply:
we should probably refactor the vpn logic in general, it's overly confusing as is